src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs

```diff
@@ -225,16 +227,22 @@ bool HasMoreToSync(out BlockHeader[]? headers, out int headersToRequest)
                     {
                         // covering edge case during fastSyncTransition when we're trying to SuggestBlock without the state
                         bool headIsGenesis = _blockTree.Head?.IsGenesis ?? false;
-                        bool toBeProcessedIsNotBlockOne = currentBlock.Number > 1;
-                        bool isFastSyncTransition = headIsGenesis && toBeProcessedIsNotBlockOne;
+                        bool toBeProcessedIsABlockWithNoState = currentBlock.Number > 1;
+                        bool isFastSyncTransition = headIsGenesis && toBeProcessedIsABlockWithNoState;
                         if (isFastSyncTransition)
                         {
-                            long bestFullState = _fullStateFinder.FindBestFullState();
+                            bestFullState ??= _fullStateFinder.FindBestFullState();
                             shouldProcess = currentBlock.Number > bestFullState && bestFullState != 0;
                             if (!shouldProcess)
                             {
                                 if (_logger.IsInfo) _logger.Info($"Skipping processing during fastSyncTransition, currentBlock: {currentBlock}, bestFullState: {bestFullState}");
-                                downloadReceipts = true;
+                                if (!downloadReceipts)
+                                {
+                                    downloadReceipts = true;
```

Comment:
We change our mind and decide we need receipts, but they have not been not loaded so we request them